### DISCUSSION TITLE
Viewer State Management

### DISCUSSION THREAD

**Discussion by joeyflora** [iTwin/itwinjs-core#5439] [Q&A]
I have multiple custom widgets in an iTwin web viewer that need to share data. What are best practices for maintaining Viewer state and sharing data between widget components?

**Comment by pratikshan85** [comment]
@joeyflora Can you please elaborate on the issue in more detail with an example?
Please mention also what kind of data you want to share among the widget components.

**Reply by joeyflora** [reply]
@pratikshan85  Sure thing. We have a table component that displays a list of elements belonging to a custom Item Type class stored as a list of flat objects with 11 properties so the data itself is not very complex. We then have another component that needs to consume the data from the table to process and export it. We would like the ability to maintain state of filtered data and selected items in the table so it can be used in the export and to update the View. I initially achieved this using React's Context API and wrapping the Viewer component with a Context Provider but have found the entire Viewer component is re-rendered when the Context Value is updated.

**Reply by pratikshan85** [reply]
@aruniverse  Can you please have a look into the issue?

**Reply by pratikshan85** [reply]
@aruniverse did you get a chance to look into the issue?
